### PR TITLE
Update useTreeData move function to allow moving item to root

### LINKED ISSUE #4074: useTreeData move does not allow moving into or within root
# 🐛 Bug Report 

The `move()` function from `useTreeData` can only move items to/within a parent item. The `insert()`, `append()`, and `prepend()` functions allow the user to specify `null` as the `parentKey` if they want to add an item to the root so I believe `move()` should follow that same pattern

## 🤔 Expected Behavior

There should be a way to move an item to or within the root of a tree created with `useTreeData`

## 😯 Current Behavior

There is no way to move an item to or within the root of a tree created with `useTreeData`

## 💁 Possible Solution

Passing `null` as the second argument to `move()` should move the item to the root, following the pattern established by `insert()`, `append()`, and `prepend()`. I'd be happy to submit a PR to fix this.

## 🔦 Context

Using this to create a nested drag-and-drop canvas and I need to be able to move items between the root and children

## 💻 Code Sample

```javascript
const tree = useTreeData({
  initialItems: [
    { 
      name: 'a', 
      items: [
        { name: 'b', items: [] }
      ] 
    },
    { 
      name: 'c', 
      items: [] 
    },
    { 
      name: 'd', 
      items: [
        { name: 'e', 
          items: [
            { name: 'f', items: [] }
          ] 
        }
      ] 
    }
  ],
  getKey: (item) => item.name,
  getChildren: (item) => item.items
});

tree.move('e', 'a', 0) // works as expected: moves item 'e' to the items array of item 'a'
tree.move('e', null, 0) // does not works as expected: I would expect this to move item 'e' to the root items array. instead the item is removed entirely. 
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-stately   | 3.20.0

### PR DESCRIPTION
**Pull request by michaeldfoley**
Closes #4074

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Verify new test in `useTreeData.test.js` passes:
```
it('should move an item to the root', function () {
  ...
}
```

### CODE REVIEW

**Comment by snowystinger** [comment]
Hey, could you follow the instructions at https://react-spectrum.adobe.com/contribute.html#contributor-license-agreement
then close and reopen the PR for it to take effect?

Thanks for the PR!

**Review by snowystinger — APPROVED** [review]

File: packages/@react-stately/data/src/useTreeData.ts [hunk 1]
```diff
@@ -322,6 +322,16 @@ export function useTreeData<T extends object>(options: TreeOptions<T>): TreeData
           parentKey: toParentKey
         };
 
+        // If parentKey is null, insert into the root.
+        if (toParentKey == null) {
```
Comment (ktabors) [inline packages/@react-stately/data/src/useTreeData.ts:326]: Do we need to support an item changing position within the root, like it's the last item out of three items and we want it to be the first item in the root? `move({name: 'Adam', parentKey: null}, null, 0)` from this tree `[{'David'}, {'John'}, {'Adam'}]`

**Comment by ktabors** [comment]
Please reopen this when the CLA is signed.

**Comment by akeohanePing** [comment]
any update on this?

**Comment by LFDanLu** [comment]
No update so far, still awaiting a signed CLA.

**Comment by akeohanePing** [comment]
@LFDanLu If I replicate the change set and open an MR, would I be able to sign the CLA?

**Comment by michaeldfoley** [comment]
@akeohanePing sorry I ended up going in a different direction for my own project and wasn't getting pinged about comments on this PR. I'll sign the CLA and reopen the PR tonight.

**Comment by snowystinger** [comment]
@akeohanePing feel free to also make the change, we can merge whichever is ready first and close the other one

**Comment by michaeldfoley** [comment]
Update PR #5277

**Comment by akeohanePing** [comment]
oh awesome thanks.

**Pull request closed without merging**